authn.py

```diff
@@ -0,0 +1,144 @@
+import os
+from dataclasses import dataclass
+
+from flask import g, request
+from functools import wraps
+from jose import jwt
+
+from typing import Dict, List, Union
+from urllib.parse import urlparse
+
+import requests
+
+
+OidcConfig = Dict[str, Union[str, List[str]]]
+Key = Dict[str, str]
+KeySet = Dict[str, List[Key]]
+
+@dataclass(frozen=True)
+class AuthError(Exception):
+    description: str
+    code: int
+
+def get_token_auth_header():
+    """Obtains the Access Token from the Authorization Header
+    """
+    auth = request.headers.get("Authorization", None)
+    if not auth:
+        raise AuthError("Authorization header is expected", 401)
+
+    parts = auth.split()
+
+    if parts[0].lower() != "bearer":
+        raise AuthError("Authorization header must start with Bearer", 401)
+    elif len(parts) == 1:
+        raise AuthError("Token not found", 401)
+    elif len(parts) > 2:
+        raise AuthError("Authorization header must be Bearer token", 401)
+
+    token = parts[1]
+    return token
+
+def requires_auth(f):
+    """Determines if the Access Token is valid
+    """
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        missing_variables = []
```

Comment:
You can move the block that initializes variables from env vars outside of the decorated function (still within `requires_auth`). We don't expect environment variables to change between invocations.